### PR TITLE
Implement end to end `CurrencyFormatter` for short and narrow formatting

### PR DESCRIPTION
**Pull request by younies**
Implements `CurrencyFormatter` to handle 'short' and 'narrow' currency formatting patterns.

### CODE REVIEW

File: components/experimental/src/dimension/format.rs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    use crate::dimension::{CurrencyCode, CurrencyFormatter};
+    use fixed_decimal::{FixedDecimal, FloatPrecision};
+
+    #[test]
+    pub fn test_es_mx() {
+        let locale = locale!("es-MX").into();
+        let fmt = CurrencyFormatter::try_new(&locale, Default::default()).unwrap();
+        let value = FixedDecimal::try_from("12345.67").unwrap();
+        let value = FixedDecimal::from("12345.67");
+        let value = FixedDecimal::try_from_f64(12345.67, FloatPrecision::Floating).unwrap();
```
Comment (younies) [inline components/experimental/src/dimension/format.rs:94]: @sffc , why there is no way to create `FixedDecimal` in tests?

Comment (younies) [reply components/experimental/src/dimension/format.rs]: Thanks for the illustration, I have used `parse`

File: components/experimental/src/dimension/format.rs [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    #[test]
+    pub fn test_en_us() {
+        let locale = locale!("en-US").into();
+        let fmt = CurrencyFormatter::try_new(&locale, Default::default()).unwrap();
+        let value = "12345.67".parse().unwrap();
+        let currency_code = CurrencyCode(tinystr!(3, "USD"));
+        let formatted_currency = fmt.format_fixed_decimal(&value, currency_code);
+        let mut sink = String::new();
+        formatted_currency.write_to(&mut sink).unwrap();
+        assert_eq!(sink.as_str(), "$12345.67");
```
Comment (sffc) [inline components/experimental/src/dimension/format.rs:97]: Please file an issue about loading grouping separators and using fixeddecimalformatter

File: components/experimental/src/dimension/options.rs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+#[derive(Debug, Eq, PartialEq, Clone, Copy)]
+#[non_exhaustive]
+pub enum Width {
+    /// Represents the short currency format.
+    /// For example, "$1.00" in English. (TODO: is this correct?)
+    Short,
+
+    /// Represents the narrow currency format.
+    /// For example, "$1" in English. (TODO: is this correct?)
+    Narrow,
```
Comment (sffc) [inline components/experimental/src/dimension/options.rs:31]:
```suggestion
pub enum Width {
    /// Format the currency with the standard (short) currency symbol.
    ///
    /// For example, 1 USD formats as "$1.00" in en-US and "US$1" in most other locales.
    Short,

    /// Format the currency with the narrow currency symbol.
    ///
    /// The narrow symbol may be ambiguous, so it should be evident from context which
    /// currency is being represented.
    ///
    /// For example, 1 USD formats as "$1.00" in most locales.
    Narrow,
```

**Review by sffc — DISMISSED** [review]

File: components/experimental/src/dimension/format.rs (hunk 2 above)
Comment (younies) [reply components/experimental/src/dimension/format.rs:97]: https://github.com/unicode-org/icu4x/issues/4740

**Review by sffc — APPROVED** [review]

**Pull request merged**